dbms/src/IO/Compression/CompressionFactory.h

```diff
@@ -74,17 +74,8 @@ class CompressionFactory
     // Create codec for compressing/decompressing data with specified settings.
     static CompressionCodecPtr create(const CompressionSettings & settings)
     {
-        if (settings.settings.size() > 1)
-        {
-            Codecs codecs;
-            codecs.reserve(settings.settings.size());
-            for (const auto & setting : settings.settings)
-            {
-                codecs.push_back(create(setting));
-            }
-            return std::make_shared<CompressionCodecMultiple>(std::move(codecs));
-        }
-        return create(settings.settings.front());
+        return settings.settings.size() > 1 ? std::make_unique<CompressionCodecMultiple>(createCodecs(settings))
+                                            : create(settings.settings.front());
```

Comment:
check that the size of `settings` must not be zero is better. Or this line may crash